python_modules/dagster/dagster/_core/definitions/remote_asset_graph.py

```diff
@@ -642,15 +642,20 @@ def split_entity_keys_by_repository(
     def build(cls, workspace: WorkspaceSnapshot):
         # Combine repository scoped asset graphs with additional context to form the global graph
 
-        code_locations = (
-            location_entry.code_location
-            for location_entry in workspace.code_location_entries.values()
-            if location_entry.code_location
+        code_locations = sorted(
+            (
+                location_entry.code_location
+                for location_entry in workspace.code_location_entries.values()
+                if location_entry.code_location
+            ),
+            key=lambda code_location: code_location.name,
         )
         repos = (
             repo
             for code_location in code_locations
-            for repo in code_location.get_repositories().values()
+            for repo in sorted(
+                code_location.get_repositories().values(), key=lambda repo: repo.name
+            )
```

Comment:
this part is to make certain that the same code location and/or repo always wins when this happens, with ties broken consistently but somewhat arbitrarily